include/alpaka/mem/buf/oacc/Copy.hpp

```diff
@@ -48,24 +48,29 @@ namespace alpaka
                     typename TTCopyPred>
                 class TTask,
                 typename TDim,
-                typename TViewDst,
+                typename TViewDstFwd,
                 typename TViewSrc,
                 typename TExtent,
                 typename TCopyPred>
             auto makeTaskCopyOacc(
-                TViewDst& viewDst,
+                TViewDstFwd&& viewDst,
                 TViewSrc const& viewSrc,
                 TExtent const& extent,
                 DevOacc const& dev,
                 TCopyPred copyPred)
             {
-                return TTask<TDim, TViewDst, TViewSrc, TExtent, TCopyPred>(viewDst, viewSrc, extent, dev, copyPred);
+                return TTask<TDim, std::remove_reference_t<TViewDstFwd>, TViewSrc, TExtent, TCopyPred>(
```

Comment:
Could `TViewDstFwd` possibly be a reference type here? If so, in which situation? (and there are a few similar cases in this PR)

Reply:
If any calling code passes an lvalue `viewDst` to `makeTaskCopyOacc`, the type of `TViewDstFwd` will be an lvalue refrence.

Reply:
I suffixed all forwarding references with `Fwd`. So these template parameters can hold either a `T` or a `T&` (which reference collapses to `T&&` and `T&`).

Reply:
Thanks for explanations!